src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java

```diff
@@ -0,0 +1,257 @@
+package org.jenkinsci.plugins.mesos;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.jvnet.hudson.test.JenkinsRule.WebClient;
+
+/**
+ * A simple client that hacks around the Jenkins configure form submit.
+ *
+ * <p>It allows to add a simple Mesos Cloud with one label.
+ */
+public class JenkinsConfigClient {
+  final OkHttpClient client;
+  final URL jenkinsConfigUrl;
+
+  public JenkinsConfigClient(WebClient jenkinsClient) throws IOException {
+    this.client = new OkHttpClient();
+    this.jenkinsConfigUrl = jenkinsClient.createCrumbedUrl("configSubmit");
+  }
+
+  /**
+   * Submits a Jenkins configuration form and adds a Mesos Cloud with one agent specs.
+   *
+   * @param mesosMasterUrl The URL of the Mesos master to connect to.
+   * @param frameworkName The Mesos framework name the plugin should use.
+   * @param role
+   * @param agentUser
+   * @param jenkinsUrl The Jenkins URL used as the base for Mesos tasks to download the Jenkins
+   *     agent.
+   * @param label The Jenkins node label of the Mesos task.
+   * @param mode Jenkins mode, can be "NORMAL" or "EXCLUSIVE".
+   * @return A {@link Response} of the {@link OkHttpClient} request.
+   * @throws IOException
+   * @throws UnsupportedEncodingException
+   */
+  public Response addMesosCloud(
+      String mesosMasterUrl,
+      String frameworkName,
+      String role,
+      String agentUser,
+      String jenkinsUrl,
+      String label,
+      String mode)
+      throws IOException, UnsupportedEncodingException {
+    final String jsonData =
```

Comment:
Do you want to leave a json example of how it should look like at the end as a comment?